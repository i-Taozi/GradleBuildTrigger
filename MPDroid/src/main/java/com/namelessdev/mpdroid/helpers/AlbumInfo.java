/*
 * Copyright (C) 2004 Felipe Gustavo de Almeida
 * Copyright (C) 2010-2014 The MPDroid Project
 *
 * All Rights Reserved.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions are met:
 *
 * 1. Redistributions of source code must retain the above copyright
 *    notice,this list of conditions and the following disclaimer.
 * 2. Redistributions in binary form must reproduce the above copyright
 *    notice, this list of conditions and the following disclaimer in the
 *    documentation and/or other materials provided with the distribution.
 *
 * THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND
 * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
 * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
 * DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE FOR ANY
 * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
 * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
 * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
 * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
 * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */

package com.namelessdev.mpdroid.helpers;

import org.a0z.mpd.Tools;
import org.a0z.mpd.item.Album;
import org.a0z.mpd.item.Artist;
import org.a0z.mpd.item.Music;

import java.util.Arrays;

import static org.a0z.mpd.Tools.getHashFromString;

public class AlbumInfo {

    private static final String INVALID_ALBUM_KEY = "INVALID_ALBUM_KEY";

    protected final String mAlbum;

    protected final String mArtist;

    protected String mFilename;

    protected String mPath;

    public AlbumInfo(final Music music) {
        super();
        String artist = music.getAlbumArtist();
        if (artist == null) {
            artist = music.getArtist();
        }
        mArtist = artist;
        mAlbum = music.getAlbum();
        mPath = music.getPath();
        mFilename = music.getFilename();
    }

    public AlbumInfo(final Album album) {
        super();

        final Artist artistName = album.getArtist();
        if (artistName != null) {
            mArtist = artistName.getName();
        } else {
            mArtist = null;
        }

        mAlbum = album.getName();
        mPath = album.getPath();
    }

    public AlbumInfo(final AlbumInfo albumInfo) {
        this(albumInfo.mArtist, albumInfo.mAlbum, albumInfo.mPath, albumInfo.mFilename);
    }

    public AlbumInfo(final String artist, final String album) {
        super();
        mArtist = artist;
        mAlbum = album;
    }

    public AlbumInfo(final String artist, final String album, final String path,
            final String filename) {
        super();
        mArtist = artist;
        mAlbum = album;
        mPath = path;
        mFilename = filename;
    }

    @Override
    public boolean equals(final Object o) {
        Boolean isEqual = null;

        if (this == o) {
            isEqual = Boolean.TRUE;
        } else if (o == null || getClass() != o.getClass()) {
            isEqual = Boolean.FALSE;
        }

        if (isEqual == null || isEqual.equals(Boolean.TRUE)) {
            final AlbumInfo albumInfo = (AlbumInfo) o;

            if (Tools.isNotEqual(mAlbum, albumInfo.mAlbum)) {
                isEqual = Boolean.FALSE;
            }

            if (Tools.isNotEqual(mArtist, albumInfo.mArtist)) {
                isEqual = Boolean.FALSE;
            }
        }

        if (isEqual == null) {
            isEqual = Boolean.TRUE;
        }

        return isEqual.booleanValue();
    }

    public String getAlbum() {
        return mAlbum;
    }

    public String getArtist() {
        return mArtist;
    }

    public String getFilename() {
        return mFilename;
    }

    public String getKey() {
        return isValid() ? getHashFromString(mArtist + mAlbum) : INVALID_ALBUM_KEY;
    }

    public String getPath() {
        return mPath;
    }

    @Override
    public int hashCode() {
        return Arrays.hashCode(new Object[]{mArtist, mAlbum});
    }

    public boolean isValid() {
        final boolean isArtistEmpty = mArtist == null || mArtist.isEmpty();
        final boolean isAlbumEmpty = mAlbum == null || mAlbum.isEmpty();
        return !isAlbumEmpty && !isArtistEmpty;
    }

    public void setFilename(final String filename) {
        mFilename = filename;
    }

    public void setPath(final String path) {
        mPath = path;
    }

    @Override
    public String toString() {
        return "AlbumInfo{" +
                "artist='" + mArtist + '\'' +
                ", album='" + mAlbum + '\'' +
                ", path='" + mPath + '\'' +
                ", filename='" + mFilename + '\'' +
                '}';
    }
}
