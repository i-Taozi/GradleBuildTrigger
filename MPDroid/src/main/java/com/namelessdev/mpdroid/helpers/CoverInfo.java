/*
 * Copyright (C) 2010-2014 The MPDroid Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.namelessdev.mpdroid.helpers;

import com.namelessdev.mpdroid.cover.ICoverRetriever;

import android.graphics.Bitmap;

public class CoverInfo extends AlbumInfo {

    public static final int MAX_SIZE = 0;

    private int mCoverMaxSize = MAX_SIZE;

    private int mCachedCoverMaxSize = MAX_SIZE;

    private Bitmap[] mBitmap = new Bitmap[0];

    private byte[] mCoverBytes = new byte[0];

    private ICoverRetriever mCoverRetriever;

    private CoverDownloadListener mListener;

    private boolean mPriority;

    private boolean mRequestGivenUp = false;

    private STATE mState = STATE.NEW;

    public CoverInfo(final AlbumInfo albumInfo) {
        super(albumInfo.getArtist(), albumInfo.getAlbum(),
                albumInfo.getPath(), albumInfo.getFilename());
    }

    public CoverInfo(final CoverInfo coverInfo) {
        super(coverInfo.mArtist, coverInfo.mAlbum, coverInfo.mPath, coverInfo.mFilename);
        mState = coverInfo.mState;
        mBitmap = coverInfo.mBitmap;
        mCoverBytes = coverInfo.mCoverBytes;
        mPriority = coverInfo.mPriority;
        mCoverMaxSize = coverInfo.mCoverMaxSize;
        mCachedCoverMaxSize = coverInfo.mCachedCoverMaxSize;
        mCoverRetriever = coverInfo.mCoverRetriever;
        mListener = coverInfo.mListener;
        mRequestGivenUp = coverInfo.mRequestGivenUp;
    }

    public Bitmap[] getBitmap() {
        return mBitmap;
    }

    public int getCachedCoverMaxSize() {
        return mCachedCoverMaxSize;
    }

    public byte[] getCoverBytes() {
        return mCoverBytes;
    }

    public int getCoverMaxSize() {
        return mCoverMaxSize;
    }

    public ICoverRetriever getCoverRetriever() {
        return mCoverRetriever;
    }

    public CoverDownloadListener getListener() {
        return mListener;
    }

    public STATE getState() {
        return mState;
    }

    public boolean isPriority() {
        return mPriority;
    }

    public boolean isRequestGivenUp() {
        return mRequestGivenUp;
    }

    public void setBitmap(final Bitmap[] bitmap) {
        mBitmap = bitmap;
    }

    public void setCachedCoverMaxSize(final int cachedCoverMaxSize) {
        mCachedCoverMaxSize = cachedCoverMaxSize;
    }

    public void setCoverBytes(final byte[] coverBytes) {
        mCoverBytes = coverBytes;
    }

    public void setCoverMaxSize(final int coverMaxSize) {
        mCoverMaxSize = coverMaxSize;
    }

    public void setCoverRetriever(final ICoverRetriever coverRetriever) {
        mCoverRetriever = coverRetriever;
    }

    @Override
    public void setFilename(final String filename) {
        mFilename = filename;
    }

    public void setListener(final CoverDownloadListener listener) {
        mListener = listener;
    }

    @Override
    public void setPath(final String path) {
        mPath = path;
    }

    public void setPriority(final boolean priority) {
        mPriority = priority;
    }

    public void setRequestGivenUp(final boolean requestGivenUp) {
        mRequestGivenUp = requestGivenUp;
    }

    public void setState(final STATE state) {
        mState = state;
    }

    @Override
    public String toString() {
        return "CoverInfo{state=" + mState + ", artist=" + mArtist == null ? ""
                : mArtist + ", album="
                        + mAlbum == null ? "" : mAlbum + " priority=" + mPriority + '}';
    }

    public enum STATE {
        NEW, CACHE_COVER_FETCH, WEB_COVER_FETCH, CREATE_BITMAP, COVER_FOUND, COVER_NOT_FOUND
    }
}
