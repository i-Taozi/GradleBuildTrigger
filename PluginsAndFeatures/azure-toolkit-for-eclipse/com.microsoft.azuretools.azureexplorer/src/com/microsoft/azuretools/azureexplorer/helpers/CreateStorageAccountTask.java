/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 */

package com.microsoft.azuretools.azureexplorer.helpers;


import com.microsoft.azure.toolkit.lib.Azure;
import com.microsoft.azure.toolkit.lib.common.model.ResourceGroup;
import com.microsoft.azure.toolkit.lib.resource.AzureGroup;
import com.microsoft.azure.toolkit.lib.resource.task.CreateResourceGroupTask;
import com.microsoft.azure.toolkit.lib.storage.model.StorageAccountConfig;
import com.microsoft.azure.toolkit.lib.storage.service.AzureStorageAccount;
import com.microsoft.azure.toolkit.lib.storage.service.StorageAccount;
import com.microsoft.azuretools.ActionConstants;
import com.microsoft.azuretools.telemetry.TelemetryConstants;
import com.microsoft.azuretools.telemetrywrapper.ErrorType;
import com.microsoft.azuretools.telemetrywrapper.EventType;
import com.microsoft.azuretools.telemetrywrapper.EventUtil;
import com.microsoft.azuretools.telemetrywrapper.Operation;
import com.microsoft.azuretools.telemetrywrapper.TelemetryManager;

import java.util.Collections;

public class CreateStorageAccountTask {

    private final StorageAccountConfig config;

    public CreateStorageAccountTask(StorageAccountConfig config) {
        this.config = config;
    }

    public StorageAccount execute() {
        final Operation operation = TelemetryManager.createOperation(ActionConstants.StorageAccount.CREATE);
        try {
            operation.start();
            final String subscriptionId = config.getSubscription().getId();
            EventUtil.logEvent(EventType.info, operation, Collections.singletonMap(TelemetryConstants.SUBSCRIPTIONID, subscriptionId));
            // create resource group if necessary.
            new CreateResourceGroupTask(subscriptionId, config.getResourceGroup().getName(), config.getRegion()).execute();

            // create storage account
            return Azure.az(AzureStorageAccount.class).create(config).commit();
        } catch (final RuntimeException e) {
            EventUtil.logError(operation, ErrorType.systemError, e, null, null);
            throw e;
        } finally {
            operation.complete();
        }
    }

}
