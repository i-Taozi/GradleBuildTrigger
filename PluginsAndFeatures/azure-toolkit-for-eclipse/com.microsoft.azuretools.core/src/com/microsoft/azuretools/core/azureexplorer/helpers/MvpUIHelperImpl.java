/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 */

package com.microsoft.azuretools.core.azureexplorer.helpers;

import com.microsoft.azuretools.core.Activator;
import com.microsoft.azuretools.core.mvp.ui.base.MvpUIHelper;

import org.eclipse.jface.dialogs.MessageDialog;

public class MvpUIHelperImpl implements MvpUIHelper {

    @Override
    public void showError(String message) {
        MessageDialog.openError(null, message, message);
    }

    @Override
    public void showException(String message, Exception e) {
        Activator.getDefault().log(message, e);
        MessageDialog.openError(null, message, e.getMessage());
    }

}
