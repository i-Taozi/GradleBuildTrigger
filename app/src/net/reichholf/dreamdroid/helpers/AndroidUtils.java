package net.reichholf.dreamdroid.helpers;

import android.content.Context;
import android.content.pm.PackageManager;
import android.view.View;
import android.widget.TextView;

public class AndroidUtils {
	public static boolean isLeanback(Context ctx) {
		return (ctx.getPackageManager().hasSystemFeature(PackageManager.FEATURE_TELEVISION)
				|| ctx.getPackageManager().hasSystemFeature(PackageManager.FEATURE_LEANBACK));
	}

	public static void setTextOrHide(TextView textView, String text) {
		if (text.isEmpty())
			textView.setVisibility(View.GONE);
		else
			textView.setText(text);
	}
}
