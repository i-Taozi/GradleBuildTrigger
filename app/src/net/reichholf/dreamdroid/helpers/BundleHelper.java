/* Â© 2010 Stephan Reichholf <stephan at reichholf dot net>
 * 
 * Licensed under the Create-Commons Attribution-Noncommercial-Share Alike 3.0 Unported
 * http://creativecommons.org/licenses/by-nc-sa/3.0/
 */

package net.reichholf.dreamdroid.helpers;

import java.util.ArrayList;

/**
 * @author sre
 *
 */
public class BundleHelper {
	public static ArrayList<String> toStringArrayList(CharSequence[] strings){
		ArrayList<String> list = new ArrayList<>();
		for (CharSequence string : strings) {
			list.add(string.toString());
		}
		return list;
	}
	
	public static CharSequence[] toCharSequenceArray(ArrayList<String> strings){
		CharSequence[] list = new CharSequence[strings.size()];
		for(int i = 0; i < strings.size(); ++i){
			list[i] = strings.get(i);
		}
		return list;
	}
}
