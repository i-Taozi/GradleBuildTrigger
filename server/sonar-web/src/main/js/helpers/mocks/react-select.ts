/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

import { GroupTypeBase, OptionProps } from 'react-select';

export function mockReactSelectOptionProps<
  OptionType,
  IsMulti extends boolean,
  GroupType extends GroupTypeBase<OptionType> = GroupTypeBase<OptionType>
>(
  data: OptionType,
  overrides?: OptionProps<OptionType, IsMulti, GroupType>
): OptionProps<OptionType, IsMulti, GroupType> {
  return {
    ...overrides,
    data
  } as OptionProps<OptionType, IsMulti, GroupType>;
}
