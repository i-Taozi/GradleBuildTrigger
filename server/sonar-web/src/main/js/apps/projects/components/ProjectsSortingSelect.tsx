/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { omit, sortBy } from 'lodash';
import * as React from 'react';
import { components, OptionProps } from 'react-select';
import { colors } from '../../../app/theme';
import { ButtonIcon } from '../../../components/controls/buttons';
import Select from '../../../components/controls/Select';
import Tooltip from '../../../components/controls/Tooltip';
import SortAscIcon from '../../../components/icons/SortAscIcon';
import SortDescIcon from '../../../components/icons/SortDescIcon';
import { translate } from '../../../helpers/l10n';
import { parseSorting, SORTING_LEAK_METRICS, SORTING_METRICS } from '../utils';

interface Props {
  className?: string;
  defaultOption: string;
  onChange: (sort: string, desc: boolean) => void;
  selectedSort: string;
  view: string;
}

export interface Option {
  label: string;
  value: string;
  className?: string;
  short?: string;
}

export default class ProjectsSortingSelect extends React.PureComponent<Props> {
  getSorting = () => {
    const options = this.getOptions();
    const { sortDesc, sortValue } = parseSorting(this.props.selectedSort);
    return { sortDesc, value: options.find(o => o.value === sortValue) };
  };

  getOptions = () => {
    const sortMetrics = this.props.view === 'leak' ? SORTING_LEAK_METRICS : SORTING_METRICS;
    return sortBy(sortMetrics, option => (option.value === this.props.defaultOption ? 0 : 1)).map(
      option => ({
        value: option.value,
        label: translate('projects.sorting', option.value),
        className: option.class
      })
    );
  };

  handleDescToggle = () => {
    const { sortDesc, sortValue } = parseSorting(this.props.selectedSort);
    this.props.onChange(sortValue, !sortDesc);
  };

  handleSortChange = (option: Option) => {
    this.props.onChange(option.value, this.getSorting().sortDesc);
  };

  projectsSortingSelectOption = (props: OptionProps<Option, false>) => {
    const { data, children } = props;
    return (
      <components.Option
        {...omit(props, ['children'])}
        className={`it__project-sort-option-${data.value} ${data.className}`}>
        {data.short ? data.short : children}
      </components.Option>
    );
  };

  render() {
    const { sortDesc, value } = this.getSorting();

    return (
      <div className={this.props.className}>
        <label id="aria-projects-sort">{translate('projects.sort_by')}:</label>
        <Select
          aria-labelledby="aria-projects-sort"
          className="little-spacer-left input-medium it__projects-sort-select"
          isClearable={false}
          onChange={this.handleSortChange}
          components={{
            Option: this.projectsSortingSelectOption
          }}
          options={this.getOptions()}
          isSearchable={false}
          value={value}
        />
        <Tooltip
          overlay={
            sortDesc ? translate('projects.sort_descending') : translate('projects.sort_ascending')
          }>
          <ButtonIcon
            aria-label={
              sortDesc
                ? translate('projects.sort_descending')
                : translate('projects.sort_ascending')
            }
            className="js-projects-sorting-invert spacer-left"
            color={colors.gray60}
            onClick={this.handleDescToggle}>
            {sortDesc ? <SortDescIcon className="" /> : <SortAscIcon className="" />}
          </ButtonIcon>
        </Tooltip>
      </div>
    );
  }
}
