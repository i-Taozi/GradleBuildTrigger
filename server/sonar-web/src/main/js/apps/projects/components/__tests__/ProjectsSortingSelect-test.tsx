/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { shallow } from 'enzyme';
import * as React from 'react';
import { GroupTypeBase } from 'react-select';
import { mockReactSelectOptionProps } from '../../../../helpers/mocks/react-select';
import { click } from '../../../../helpers/testUtils';
import ProjectsSortingSelect, { Option } from '../ProjectsSortingSelect';

it('should render correctly for overall view', () => {
  expect(shallowRender()).toMatchSnapshot();
});

it('should render correctly for leak view', () => {
  expect(
    shallowRender({ defaultOption: 'analysis_date', selectedSort: 'new_coverage', view: 'leak' })
  ).toMatchSnapshot();
});

it('should handle the descending sort direction', () => {
  expect(shallowRender({ selectedSort: '-vulnerability' })).toMatchSnapshot();
});

it('should render option correctly', () => {
  const wrapper = shallowRender();
  const SortOption = wrapper.instance().projectsSortingSelectOption;
  expect(
    shallow(
      <SortOption
        {...mockReactSelectOptionProps<Option, false, GroupTypeBase<Option>>({
          label: 'foo',
          value: 'foo',
          short: 'fo'
        })}
      />
    )
  ).toMatchSnapshot();
  expect(
    shallow(
      <SortOption
        {...mockReactSelectOptionProps<Option, false, GroupTypeBase<Option>>({
          label: 'foo',
          value: 'foo'
        })}
      />
    )
  ).toMatchSnapshot();
});

it('changes sorting', () => {
  const onChange = jest.fn();
  const instance = shallowRender({
    selectedSort: '-vulnerability',
    onChange
  }).instance() as ProjectsSortingSelect;
  instance.handleSortChange({ label: 'size', value: 'size' });
  expect(onChange).toBeCalledWith('size', true);
});

it('reverses sorting', () => {
  const onChange = jest.fn();
  const wrapper = shallowRender({ selectedSort: '-size', onChange });
  click(wrapper.find('ButtonIcon'));
  expect(onChange).toBeCalledWith('size', false);
});

function shallowRender(overrides: Partial<ProjectsSortingSelect['props']> = {}) {
  return shallow<ProjectsSortingSelect>(
    <ProjectsSortingSelect
      defaultOption="name"
      onChange={jest.fn()}
      selectedSort="name"
      view="overall"
      {...overrides}
    />
  );
}
