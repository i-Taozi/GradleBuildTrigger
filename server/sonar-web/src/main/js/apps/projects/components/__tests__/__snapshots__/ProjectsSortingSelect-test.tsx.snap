// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle the descending sort direction 1`] = `
<div>
  <label
    id="aria-projects-sort"
  >
    projects.sort_by
    :
  </label>
  <Select
    aria-labelledby="aria-projects-sort"
    className="little-spacer-left input-medium it__projects-sort-select"
    components={
      Object {
        "Option": [Function],
      }
    }
    isClearable={false}
    isSearchable={false}
    onChange={[Function]}
    options={
      Array [
        Object {
          "className": undefined,
          "label": "projects.sorting.name",
          "value": "name",
        },
        Object {
          "className": undefined,
          "label": "projects.sorting.analysis_date",
          "value": "analysis_date",
        },
        Object {
          "className": undefined,
          "label": "projects.sorting.reliability",
          "value": "reliability",
        },
        Object {
          "className": undefined,
          "label": "projects.sorting.security",
          "value": "security",
        },
        Object {
          "className": undefined,
          "label": "projects.sorting.security_review",
          "value": "security_review",
        },
        Object {
          "className": undefined,
          "label": "projects.sorting.maintainability",
          "value": "maintainability",
        },
        Object {
          "className": undefined,
          "label": "projects.sorting.coverage",
          "value": "coverage",
        },
        Object {
          "className": undefined,
          "label": "projects.sorting.duplications",
          "value": "duplications",
        },
        Object {
          "className": undefined,
          "label": "projects.sorting.size",
          "value": "size",
        },
      ]
    }
  />
  <Tooltip
    overlay="projects.sort_descending"
  >
    <ButtonIcon
      aria-label="projects.sort_descending"
      className="js-projects-sorting-invert spacer-left"
      color="#999"
      onClick={[Function]}
    >
      <SortDescIcon
        className=""
      />
    </ButtonIcon>
  </Tooltip>
</div>
`;

exports[`should render correctly for leak view 1`] = `
<div>
  <label
    id="aria-projects-sort"
  >
    projects.sort_by
    :
  </label>
  <Select
    aria-labelledby="aria-projects-sort"
    className="little-spacer-left input-medium it__projects-sort-select"
    components={
      Object {
        "Option": [Function],
      }
    }
    isClearable={false}
    isSearchable={false}
    onChange={[Function]}
    options={
      Array [
        Object {
          "className": undefined,
          "label": "projects.sorting.analysis_date",
          "value": "analysis_date",
        },
        Object {
          "className": undefined,
          "label": "projects.sorting.name",
          "value": "name",
        },
        Object {
          "className": "projects-leak-sorting-option",
          "label": "projects.sorting.new_reliability",
          "value": "new_reliability",
        },
        Object {
          "className": "projects-leak-sorting-option",
          "label": "projects.sorting.new_security",
          "value": "new_security",
        },
        Object {
          "className": "projects-leak-sorting-option",
          "label": "projects.sorting.new_security_review",
          "value": "new_security_review",
        },
        Object {
          "className": "projects-leak-sorting-option",
          "label": "projects.sorting.new_maintainability",
          "value": "new_maintainability",
        },
        Object {
          "className": "projects-leak-sorting-option",
          "label": "projects.sorting.new_coverage",
          "value": "new_coverage",
        },
        Object {
          "className": "projects-leak-sorting-option",
          "label": "projects.sorting.new_duplications",
          "value": "new_duplications",
        },
        Object {
          "className": "projects-leak-sorting-option",
          "label": "projects.sorting.new_lines",
          "value": "new_lines",
        },
      ]
    }
    value={
      Object {
        "className": "projects-leak-sorting-option",
        "label": "projects.sorting.new_coverage",
        "value": "new_coverage",
      }
    }
  />
  <Tooltip
    overlay="projects.sort_ascending"
  >
    <ButtonIcon
      aria-label="projects.sort_ascending"
      className="js-projects-sorting-invert spacer-left"
      color="#999"
      onClick={[Function]}
    >
      <SortAscIcon
        className=""
      />
    </ButtonIcon>
  </Tooltip>
</div>
`;

exports[`should render correctly for overall view 1`] = `
<div>
  <label
    id="aria-projects-sort"
  >
    projects.sort_by
    :
  </label>
  <Select
    aria-labelledby="aria-projects-sort"
    className="little-spacer-left input-medium it__projects-sort-select"
    components={
      Object {
        "Option": [Function],
      }
    }
    isClearable={false}
    isSearchable={false}
    onChange={[Function]}
    options={
      Array [
        Object {
          "className": undefined,
          "label": "projects.sorting.name",
          "value": "name",
        },
        Object {
          "className": undefined,
          "label": "projects.sorting.analysis_date",
          "value": "analysis_date",
        },
        Object {
          "className": undefined,
          "label": "projects.sorting.reliability",
          "value": "reliability",
        },
        Object {
          "className": undefined,
          "label": "projects.sorting.security",
          "value": "security",
        },
        Object {
          "className": undefined,
          "label": "projects.sorting.security_review",
          "value": "security_review",
        },
        Object {
          "className": undefined,
          "label": "projects.sorting.maintainability",
          "value": "maintainability",
        },
        Object {
          "className": undefined,
          "label": "projects.sorting.coverage",
          "value": "coverage",
        },
        Object {
          "className": undefined,
          "label": "projects.sorting.duplications",
          "value": "duplications",
        },
        Object {
          "className": undefined,
          "label": "projects.sorting.size",
          "value": "size",
        },
      ]
    }
    value={
      Object {
        "className": undefined,
        "label": "projects.sorting.name",
        "value": "name",
      }
    }
  />
  <Tooltip
    overlay="projects.sort_ascending"
  >
    <ButtonIcon
      aria-label="projects.sort_ascending"
      className="js-projects-sorting-invert spacer-left"
      color="#999"
      onClick={[Function]}
    >
      <SortAscIcon
        className=""
      />
    </ButtonIcon>
  </Tooltip>
</div>
`;

exports[`should render option correctly 1`] = `
<Option
  className="it__project-sort-option-foo undefined"
  data={
    Object {
      "label": "foo",
      "short": "fo",
      "value": "foo",
    }
  }
>
  fo
</Option>
`;

exports[`should render option correctly 2`] = `
<Option
  className="it__project-sort-option-foo undefined"
  data={
    Object {
      "label": "foo",
      "value": "foo",
    }
  }
/>
`;
